docs/query-builder.txt

```diff
@@ -9,588 +9,1095 @@ Query Builder
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
 
-When using MongoDB connections, you will be able to build fluent queries to
-perform database operations.
+Overview
+--------
 
-For your convenience, there is a ``collection`` alias for ``table`` and
-other MongoDB specific operators/operations.
+In this guide, you can learn how to use the {+odm-short+} extension of
+the Laravel query builder to work with a MongoDB database. The query builder
+lets you write queries for any supported database by using the same fluent
+interface and syntax.
 
-.. code-block:: php
-
-   $books = DB::collection('books')->get();
-
-   $hungerGames =
-       DB::collection('books')
-           ->where('name', 'Hunger Games')
-           ->first();
-
-If you are familiar with `Eloquent Queries <http://laravel.com/docs/queries>`__,
-there is the same functionality.
-
-Available operations
---------------------
-
-**Retrieving all models**
-
-.. code-block:: php
-
-   $users = User::all();
-
-**Retrieving a record by primary key**
+.. note::
 
-.. code-block:: php
+   {+odm-short+} extends Laravel's query builder and Eloquent ORM, which can
+   run similar database operations. To learn more about retrieving documents
+   by using Eloquent models, see :ref:`laravel-fundamentals-retrieve`.
 
-   $user = User::find('517c43667db388101e00000f');
+Laravel provides a **facade** to access the query builder class ``DB``, which
+lets you perform database operations. Facades are static interfaces to
+classes that make the syntax more concise, avoid runtime errors, and improve
+testability.
 
-**Where**
+{+odm-short+} aliases the ``DB`` method ``table()`` as the ``collection()``
+method. Chain methods to specify the command and any constraints. Then, chain
+the ``get()`` method at the end to run them on the MongoDB collection. The
+following example shows the syntax of a query builder call:
 
 .. code-block:: php
 
-   $posts =
-       Post::where('author.name', 'John')
-           ->take(10)
-           ->get();
-
-**OR Statements**
-
-.. code-block:: php
+   DB::collection('<collection name>')
+       // chain methods by using the "->" object operator
+       ->get();
 
-   $posts =
-       Post::where('votes', '>', 0)
-           ->orWhere('is_approved', true)
-           ->get();
+This guide shows examples of the following types of query builder operations:
+
+- :ref:`laravel-retrieve-query-builder`
+- :ref:`laravel-modify-results-query-builder`
+- :ref:`laravel-mongodb-read-query-builder`
+- :ref:`laravel-mongodb-write-query-builder`
+
+Before You Get Started
+----------------------
+
+To run the code examples in this guide, complete the
+:ref:`Quick Start <laravel-quick-start>` tutorial to configure a web
+application, load sample datasets into your MongoDB deployment, and
+run the example code from a controller method.
+
+To perform read and write operations by using the query builder, import the
+``Illuminate\Support\Facades\DB`` facade and compose your query.
```

Comment:
Good question! Based on feedback in a prior PR from @GromNaN, the developer would be using an IDE which would auto-suggest the imports. In this case, although it might get auto-suggested, I think it's helpful to mention the import package to help the reader know where it came from.